framework/web/Request.php

```diff
@@ -1097,8 +1187,8 @@ public function getContentType()
     public function getAcceptableLanguages()
     {
         if ($this->_languages === null) {
-            if (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
-                $this->_languages = array_keys($this->parseAcceptHeader($_SERVER['HTTP_ACCEPT_LANGUAGE']));
+            if ($this->headers->get('Accept-Language')) {
```

Comment:
may be faster in general to call get and check for null. Also in other places where isset and `$_SERVER` have been replaced.